src/app/config/config.c

```diff
@@ -7653,8 +7653,7 @@ get_first_listener_addrport_string(int listener_type)
     if (cfg->server_cfg.no_listen)
       continue;
 
-    if (cfg->type == listener_type &&
-        tor_addr_family(&cfg->addr) != AF_UNSPEC) {
+    if (cfg->type == listener_type && tor_addr_is_unspec(&cfg->addr)) {
```

Comment:
This change tests for unspec, it used to test for not unspec.
Maybe you should use an automated tool like coccinelle for this commit?

Reply:
I fixed this and pushed it.
For future tickets similar to this (in terms of refactoring), I will look into coccinelle.